insteon-scene-explorer/ise_pack.yaml

```diff
@@ -0,0 +1,289 @@
+      condition: []
+      action:
+        - service: input_number.set_value
+          data_template:
+            entity_id: input_number.target_insteon_group
+            value: "{{ states('input_select.insteon_modem_groups') | int }}"
+        - service: homeassistant.update_entity
+          data: {}
+          target:
+            entity_id:
+            - sensor.insteon_group_names
+      mode: single
+
+    #################################
+    ## Refresh "sensor.insteon_group_names"
+    #################################
+    - id: "1658021777"
+      alias: ISE - Change Named Groups
+      description: ""
+      trigger:
+        - platform: state
+          entity_id: sensor.insteon_named_groups
+          attribute: groups
+        - platform: homeassistant
+          event: start
+      condition: []
+      action:
+        - service: homeassistant.update_entity
+          data: {}
+          target:
+            entity_id:
+            - sensor.insteon_group_names
+      mode: single
+
+    #################################
+    ## Iterate over all Insteon Scenes and create Home Assistant Groups that match.
+    #################################
+    - id: "1655407823753"
+      alias: ISE - Build Insteon Scenes
+      description: ""
+      trigger:
+        - platform: state
+          entity_id: sensor.insteon_groups
+          attribute: groups
+        - platform: homeassistant
+          event: start
+      condition: []
+      action:
+        - repeat:
+            while:
+              - condition: template
+                value_template: >-
+                  {{ repeat.index <= (state_attr('input_select.insteon_modem_groups','options') | count) }}
+            sequence:
+              - service: group.set
+                data_template:
+                  object_id: >-
+                    {{ 'insteon_group_' + (state_attr('input_select.insteon_modem_groups','options')[repeat.index - 1]) }}
+                  name: >-
+                    {% set scene_number = (state_attr('input_select.insteon_modem_groups','options')[repeat.index - 1]) | string %} 
+                    {% set mapper = state_attr('sensor.insteon_named_groups','groups') %}
+                    {% set gstring = namespace(grname=[]) %}  
+                    {% set gstring.grname = mapper[scene_number] + ' (' + scene_number + ')' if scene_number in mapper else 'Insteon Group ' + scene_number %}
+                    {{ gstring.grname }}
+                  icon: mdi:lightbulb-group
+                  entities: >-
+                    {% set glist = state_attr('sensor.insteon_groups','groups') %}  
+                    {% set devices = namespace(switch=[]) %}  
+                    {% for value in glist %} 
+                    {%   if ((value.group | int) == (state_attr('input_select.insteon_modem_groups','options')[repeat.index - 1]) | int)  %} 
+                    {%     set devices.switch = devices.switch + [value.device_fqn | string | upper] %}  
```

Comment:
notice the simplified code because it uses the device_fqn property introduced earlier.